types/lucene/index.d.ts

```diff
@@ -10,20 +10,37 @@ export interface TermLocation {
     offset: number;
 }
 
-export interface Node {
-    boost: null | number;
+export interface NodeField {
     field: string | '<implicit>';
-    fieldLocation: null;
+    fieldLocation: null | {
+        end: TermLocation;
+        start: TermLocation;
+    };
+}
+
+export type NodeTerm = NodeField & {
+    boost: null | number;
     prefix: null | string;
     quoted: boolean;
     regex: boolean;
-    similarity: null;
+    similarity: null | number;
     term: string;
     termLocation: {
         start: TermLocation;
         end: TermLocation;
     };
-}
+};
+
+export type NodeRangedTerm = NodeField & {
```

Comment:
(Same as above) any reason to use a `type` and not an `interface` here (i.e. `export interface NodeRangedTerm extends NodeField { ... }`).

Using interfaces allows users of the library to augment the types in their projects, if they need to - while types don't.

Reply:
That counter argument is going to take some explaining.  I'll switch it over, but as far as I understand them there isn't a strong difference between a type and an interface other than syntax.  Sounds like a learning opportunity!